app/views/experiments/table.html.erb

```diff
@@ -23,7 +23,7 @@
           </div>
         </div>
         <% Experiments::TableViewService::COLUMNS.each do |col| %>
-          <div class="table-header-cell <%= col %>-column">
+          <div class="table-header-cell <%= col %>-column <%= 'hidden' unless col.to_s.in?(@my_module_visible_table_columns) || col == :task_name %>">
```

Comment:
You just need to refresh with javascript column state.

Reply:
Doesn't that already happens when you switch columns on and off? It seems to work OK, this is just the initial state of the table.